test/loader/BasePackageLoader.test.ts

```diff
@@ -0,0 +1,1390 @@
+        throw new Error('get package stats error');
+      });
+
+      const result = await loader.loadPackage('empty-json-id', '0.1.0');
+      expect(result).toBe(LoadStatus.FAILED);
+      // expect(loggerSpy.getLastMessage('info')).toBe('JSON file at path Package/json/path/empty-json-id/0.1.0 was not FHIR resource');
+      expect(packageCacheMock.getPotentialResourcePaths).toHaveBeenCalled();
+      expect(packageDBMock.savePackageInfo).toHaveBeenCalled();
+      expect(packageDBMock.saveResourceInfo).not.toHaveBeenCalled();
+    });
+
+    it('should use non local package paths to find package with extension flavor', async () => {
+      const resourceResourceFlavor = path.resolve(
+        __dirname,
+        'fixtures',
+        'StructureDefinition-patient-birthPlace.json'
+      );
+      const pkg = setupLoadPackage(
+        'patient-birthPlace',
+        '4.0.1',
+        'not-loaded',
+        undefined,
+        path.resolve(__dirname, 'fixtures'),
+        resourceResourceFlavor,
+        '2024-05-24T16:27:17-04:00'
+      );
+      packageCacheMock.isPackageInCache.calledWith(pkg.name, pkg.version).mockReturnValue(true);
+      packageCacheMock.getPackagePath
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue(pkg.packageJsonPath);
+      packageCacheMock.getPackageJSONPath
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue(pkg.packageJsonPath);
+      packageCacheMock.getPotentialResourcePaths
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue([pkg.resourcePath]);
+      const humanBeingJSON = fs.readJsonSync(resourceResourceFlavor);
+      packageCacheMock.getResourceAtPath
+        .calledWith(pkg.resourcePath)
+        .mockReturnValue(humanBeingJSON);
+      packageDBMock.getPackageStats
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue({ name: pkg.name, version: pkg.version, resourceCount: 1 });
+
+      const result = await loader.loadPackage('patient-birthPlace', '4.0.1');
+      expect(packageDBMock.savePackageInfo).toHaveBeenCalled();
+      expect(packageDBMock.saveResourceInfo).toHaveBeenCalled();
+      expect(packageDBMock.saveResourceInfo).toHaveBeenCalledWith({
+        resourceType: 'StructureDefinition',
+        id: 'patient-birthPlace',
+        url: 'http://hl7.org/fhir/StructureDefinition/patient-birthPlace',
+        name: 'birthPlace',
+        version: '4.0.1',
+        sdKind: 'complex-type',
+        sdDerivation: 'constraint',
+        sdType: 'Extension',
+        sdBaseDefinition: 'http://hl7.org/fhir/StructureDefinition/Extension',
+        sdAbstract: false,
+        sdFlavor: 'Extension',
+        packageName: 'patient-birthPlace',
+        packageVersion: '4.0.1',
+        resourcePath: resourceResourceFlavor
+      });
+      expect(result).toBe(LoadStatus.LOADED);
+    });
+
+    it('should show an info message when a JSON file in a loaded package is not a FHIR resource', async () => {
```

Comment:
As noted in the production code for this, I actually think this should be a `debug` message instead since it's an expected (and fairly common) situation.